src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_params.py

```diff
@@ -155,6 +155,8 @@ def load_arguments(self, _):
         c.argument('application_security_groups', resource_type=ResourceType.MGMT_NETWORK, min_api='2017-09-01', nargs='+', options_list=['--asgs'], help='Space-separated list of existing application security groups to associate with the VM.', arg_group='Network', validator=validate_asg_names_or_ids)
         c.argument('write_accelerator', nargs='*', min_api='2017-12-01', arg_group='Storage',
                    help="enable/disable disk write accelerator. Use singular value 'true/false' to apply across, or specify individual disks, e.g.'os=true 1=true 2=true' for os disk and data disks with lun of 1 & 2")
+        c.argument('boot_diagnostics_storage',
+                   help='pre-existing storage account name or its blob uri to capture boot diagnostics. Its sku should be one of Standard_GRS, Standard_LRS and Standard_RAGRS')
```

Comment:
Does the service need permission to the storage account?

Reply:
No extra permission configuration is needed. I believe CRP just accesses the storage under VM creator's identity captured through the original token from the VM PUT